### PR TITLE
Support for audio tracks in HLS video

### PR DESCRIPTION
**Pull request by egreaves**
Added support for audio tracks in the html5 provider.

### CODE REVIEW

File: src/js/providers/html5.js [hunk 1]
```diff
@@ -140,10 +140,10 @@ define([
             _beforecompleted = false,
 
             _fullscreenState = false,
-
-            // Video Text Tracks
+            // Video Tracks
             _textTracks,
-            _currentTextTrackIndex = -1;
+            _currentTextTrackIndex = -1,
+            _currentAudioTrackIndex = -1;
```
Comment (robwalch) [inline src/js/providers/html5.js]: Are these media tracks state properties being reset when loading new media?

File: src/js/providers/html5.js [hunk 2]
```diff
@@ -837,7 +838,25 @@ define([
         this.getName = function() {
             return { name : _name };
         };
+        this.setCurrentAudioTrack = _setCurrentAudioTrack;
 
+        function _setAudioTracks(tracks) {
+            if(tracks && tracks.length > 0) {
+                _currentAudioTrackIndex = 0;
+                tracks[_currentAudioTrackIndex].enabled = true;
```
Comment (robwalch) [inline src/js/providers/html5.js]: Is it necessary to enable the first track? The media might define any track as the default track, and we should not change that when getting the list of available tracks.

Comment (robwalch) [inline src/js/providers/html5.js]: So instead look for the first enabled track, and set `_currentAudioTrackIndex ` to that.

File: src/js/view/controlbar.js [hunk 3]
```diff
@@ -305,7 +305,7 @@ define([
             if (this.elements.playlist) {
                 this.elements.playlist.selectItem(itemIdx);
             }
-
+            //TO-DO: determine if this is the best spot for audio tracks setup
```
Comment (robwalch) [inline src/js/view/controlbar.js]: Is this comment necessary?

File: src/js/providers/html5.js (hunk 1 above)
Comment (egreaves) [inline src/js/providers/html5.js]: Not currently. Doing so after the loadeddata event fires should address this.

File: src/js/providers/html5.js (hunk 2 above)
Comment (egreaves) [inline src/js/providers/html5.js]: If the media item always sets a default, then this isn't necessary.

File: src/js/providers/html5.js (hunk 1 above)
Comment (robwalch) [inline src/js/providers/html5.js]: Only when there are tracks in the new source. You should clear these on init/load.

File: src/js/providers/html5.js (hunk 2 above)
Comment (robwalch) [inline src/js/providers/html5.js]: It should or enable the first by default.

File: src/js/view/controlbar.js (hunk 3 above)
Comment (egreaves) [inline src/js/view/controlbar.js]: It isn't anymore; I'll remove it.

**Pull request merged**